src/oscar/apps/basket/forms.py

```diff
@@ -19,6 +19,16 @@ def __init__(self, strategy, *args, **kwargs):
         super(BasketLineForm, self).__init__(*args, **kwargs)
         self.instance.strategy = strategy
 
+        max_allowed_quantity = None
+        num_available = getattr(self.instance.purchase_info.availability, 'num_available', None)
+        basket_max_allowed_quantity = self.instance.basket.max_allowed_quantity()[0]
+        if all([num_available, basket_max_allowed_quantity]):
+            max_allowed_quantity = min(num_available, basket_max_allowed_quantity)
+        elif any([num_available, basket_max_allowed_quantity]):
```

Comment:
This can be simplified to just `else:` - the `any` check is redundant.

Reply:
In `else` condition they both could be `None`, so I believe `any` check is necessary.

Reply:
No - because you are doing a second `if` check below. It will not add an attr if both are `None`.